doc/mediasdk-man.md

```diff
@@ -5734,7 +5734,7 @@ This structure specifies configurations for decoding, encoding and transcoding p
 `TargetUsage`           | Target usage model that guides the encoding process; see the [TargetUsage](#TargetUsage) enumerator for details.
 `RateControlMethod`     | Rate control method; see the [RateControlMethod](#RateControlMethod) enumerator for details.
 `InitialDelayInKB`, `TargetKbps`, `MaxKbps` | These parameters are for the constant bitrate (CBR), variable bitrate control (VBR) and [CQP HRD](#Appendix_F) algorithms.<br><br>The SDK encoders follow the Hypothetical Reference Decoding (HRD) model. The HRD model assumes that data flows into a buffer of the fixed size `BufferSizeInKB` with a constant bitrate `TargetKbps`. (Estimate the targeted frame size by dividing the framerate by the bitrate.)<br><br>The decoder starts decoding after the buffer reaches the initial size `InitialDelayInKB`, which is equivalent to reaching an initial delay of `InitialDelayInKB*8000/TargetKbps`ms. Note: In this context, KB is 1000 bytes and Kbps is 1000 bps.<br><br>If `InitialDelayInKB` or `BufferSizeInKB` is equal to zero, the value is calculated using bitrate, frame rate, profile, level, and so on.<br><br>`TargetKbps` must be specified for encoding initialization.<br><br>For variable bitrate control, the `MaxKbps` parameter specifies the maximum bitrate at which the encoded data enters the Video Buffering Verifier (VBV) buffer. If `MaxKbps` is equal to zero, the value is calculated from bitrate, frame rate, profile, level, and so on.
-`QPI, QPP, QPB`         | Quantization Parameters (`QP)  `for `I`, `P` and `B` frames, respectively, for the constant `QP` (CQP) mode.
+`QPI, QPP, QPB`         | Quantization Parameters (`QP`) for `I`, `P` and `B` frames for constant `QP` mode (CQP). All input non-zero QPs are clipped to QP range supported on hardware being used. Zero QP is not valid and means that some default value will be assigned by MediaSDK library. This value depends on the codec and can be changed in future MediaSDK releases. E.g. HEVC Encoder calculates default values based on the valid ones to follow the formula QPP = QPI + 2, QPB = QPP + 2. If QPI is not set the average value of QP range is used.
```

Comment:
I'd propose this :
Quantization Parameters (`QP`) for `I`, `P` and `B` frames for constant `QP` mode (CQP). Zero QP is not valid and means that a default value will be assigned by SDK. Non-zero QPs might be clipped to supported QP range. 
**Note**: Default QP values for QPI/QPP/QPB are implementation dependent and subject to change without an additional notice in this document.

Reply:
@stasefremov , @aleonenk , @eistarov , @dmitry-gurulev , can you please review this part?

Reply:
I would add detailed information about ranges for different codecs and bit depths. Maybe a table.

Reply:
>I would add detailed information about ranges for different codecs and bit depths. Maybe a table.

 @stasefremov , per codec/platform?

Reply:
Not sure if platform limitations are subject for MSDK manual.. I'm talking about fundamental ranges which depend on codec/bitdepth.

Reply:
I'm afraid that in this case we'll have to update this data pretty often. E.g. for HEVC driver minQP is 10 for VDENC and 0 for VME, starting from ICL negative QP is supported on VME and so on. Some restrictions msdk can obtain through caps. So this table can be very hard to support.
I basically like Dima's formulation

Reply:
Frequent changes would be required for platform limitations. I think fundamental ranges (e.g. 0-51 for HEVC 8-bit, 0-255 for VP9 8-bit, etc.) shouldn't be changed at all.

Reply:
@stasefremov , we don't support '0'

Reply:
Yes. This limitation (unsupported value 0) is essential part of parameter semantics. It should be excluded from "fundamental" range.

Reply:
Can we comment how AVC will behave in this case? If you will be able to cover all mediasdk supported encoders this will be really cool.

Reply:
Hey guys, are we really going to describe here all QP requirements for all codecs msdk supports? In this case we'll never have this code checked in.
Can we just say that QP range follows corresponding specification but still can be different on different platforms and to one has to use Query to get real numbers?

Reply:
> Can we comment how AVC will behave in this case? If you will be able to cover all mediasdk supported encoders this will be really cool.

It is not an approach we can maintain. I'm against mentioning implementation details in the manual.

Reply:
In this case I suggest to drop this sentence entirely: "E.g. HEVC Encoder calculates default values based on the valid ones to follow the formula QPP = QPI + 2, QPB = QPP + 2. If QPI is not set the average value of QP range is used." Without it text seems to be ok. With it you need to add info for other codecs. At least for AVC.

Reply:
We're on the same page.

Reply:
> Can we just say that QP range follows corresponding specification

AFAIK it's not the case for HEVC high bit depth.
 
> but still can be different on different platforms and to one has to use Query to get real numbers?

Unfortunately Query can't be used to get an idea about proper range with adequate efforts. You can check particular value if you already understand what full range for the parameter is.

I agree that we don't need to mention anything related to particular implementation/platform (HW/SW restrictions for particular platform, rules for deduction of unknown QP type from known QP type, etc.). But I think we need to give an idea about full/fundamental range for QP parameter for particular codec/bitdepth.